### ISSUE TITLE
Fix false positive with same method across different anonymous modules

### ISSUE THREAD

**Issue by bkuhlmann** [rubocop/rubocop#15056]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Lint/DuplicateMethods

### Steps to reproduce the problem

Create a file with the following content:

``` ruby
# demo.rb

one = Module.new { def test(**one) = super }
            .instance_method(:test)
            .parameters

puts one

two = Module.new { def test(**one) = super }
            .instance_method(:test)
            .parameters

puts two
```

Run `rubocop demo.rb` to see the output.

### Expected behavior

I would expect this to not raise a duplication error. I suppose, one could refactor the RSpec code to put the anonymous object in a `let` but the fact that this is anonymous shouldn't matter?

### Actual behavior

When you run the above, you'll get the following:

Lint/DuplicateMethods: Method Object#test is defined at both /Users/bkuhlmann/Engineering/Misc/demo:15 and /Users/bkuhlmann/Engineering/Misc/demo:21.
two = Module.new { def test(**one) = super }

This is similar to and a variant of what's been reported in #15028. 

:bulb: To provide more context, the above was extracted from my RSpec test suite where I need an anonymous module for testing purposes where, yes, the method is the same but I need to check for different parameters, different setups, etc.

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
1.86.0 (using Parser 3.3.10.2, Prism 1.9.0, rubocop-ast 1.49.1, analyzing as Ruby 4.0, running on ruby 4.0.2) [arm64-darwin25.3.0]
  - rubocop-capybara 2.22.1
  - rubocop-disable_syntax 0.2.0
  - rubocop-packaging 0.6.0
  - rubocop-performance 1.26.1
  - rubocop-thread_safety 0.7.3
  - rubocop-rake 0.7.1
  - rubocop-rspec 3.9.0
```

**Comment by koic** [comment]
It looks like #15055 resolves this issue. What do you think?

**Comment by bkuhlmann** [comment]
Ah, fantastic. Closing. Will double check when the next RuboCop patch is released. Looking forward to this!
